extension/api.js

```diff
@@ -78,149 +105,243 @@ const lostEngines = new Map ([
+          if (allow) {
+            await ExtensionSettingsStore.initialize();
+            ExtensionSettingsStore.addSetting(
+              extension.id,
+              DEFAULT_SEARCH_STORE_TYPE,
+              DEFAULT_SEARCH_SETTING_NAME,
+              engineName,
+              () => defaultEngine.name
+            );
+            Services.search.defaultEngine = Services.search.getEngineByName(
+              engineName
+            );
+          }
 
-      console.log(`reset-default-search: ask user to set default search engine to ${addon.id}`);
-      let window = BrowserWindowTracker.getTopWindow({ allowPopups: false });
-      let subject = {
-        wrappedJSObject: {
-          browser: window.gBrowser.selectedBrowser,
-          name: extension.name,
-          icon: extension.iconURL,
-          currentEngine: defaultEngine.name,
-          newEngine: engineName,
-          async respond(allow) {
-            if (allow) {
-              await ExtensionSettingsStore.initialize();
-              ExtensionSettingsStore.addSetting(
-                extension.id,
-                DEFAULT_SEARCH_STORE_TYPE,
-                DEFAULT_SEARCH_SETTING_NAME,
-                engineName,
-                () => defaultEngine.name
-              );
-              Services.search.defaultEngine = Services.search.getEngineByName(
-                engineName
-              );
-            }
-
-            try {
-              Services.telemetry.recordEvent(
-                "defaultSearchReset",
-                "interaction",
-                "panelShown",
-                extension.id
-              );
-            } catch (err) {
-              // If the telemetry throws just log the error so it doesn't break any
-              // functionality.
-              Cu.reportError(err);
-            }
-            // Remember that we have completed.
-            finish(true);
-          },
+          // Remember that we have completed.
+          finish(true, "interaction", allow ? "accepted" : "denied", extension.id);
         },
-      };
-      Services.obs.notifyObservers(
-        subject,
-        "webextension-defaultsearch-prompt"
+      },
+    };
+    Services.obs.notifyObservers(
+      subject,
+      "webextension-defaultsearch-prompt"
+    );
+
+    try {
+      Services.telemetry.recordEvent(
+        "defaultSearchReset",
+        "interaction",
+        "ask",
+        extension.id
       );
-      // We only prompt for the first addon that makes it here.  If the user does
-      // not respond to the panel, they will be asked again on next startup.
+    } catch (err) {
+      // If the telemetry throws just log the error so it doesn't break any
+      // functionality.
+      Cu.reportError(err);
+    }
+
+    // We only prompt for the first addon that makes it here.  If the user does
+    // not respond to the panel, they will be asked again on next startup.
```

Comment:
Do we want to record in telemetry if the user did not respond?

Reply:
We have no good way to determine that.  We do record that the panel is shown, then we record the response.  We'll see if there are more "asks" than responses in the data.